apps/server/src/modules/h5p-editor/service/libraryStorage.service.spec.ts

```diff
@@ -0,0 +1,658 @@
+import { Readable } from 'stream';
+
+import { HeadObjectCommandOutput, ServiceOutputTypes } from '@aws-sdk/client-s3';
+import { Test, TestingModule } from '@nestjs/testing';
+import { DeepMocked, createMock } from '@golevelup/ts-jest';
+import { ILibraryMetadata, ILibraryName } from '@lumieducation/h5p-server';
+import { NotFoundException } from '@nestjs/common';
+
+import { S3ClientAdapter } from '@src/modules/files-storage/client/s3-client.adapter';
+import { IGetFileResponse } from '@src/modules/files-storage/interface';
+
+import { LibraryRepo } from '../repo/library.repo';
+import { LibraryStorage } from './libraryStorage.service';
+import { FileMetadata, InstalledLibrary } from '../entity/library.entity';
+
+async function readStream(stream: Readable): Promise<string> {
+	// eslint-disable-next-line @typescript-eslint/no-explicit-any
+	const chunks: any[] = [];
+	return new Promise((resolve, reject) => {
+		stream.on('data', (chunk) => chunks.push(chunk));
+		stream.on('error', reject);
+		stream.on('end', () => resolve(Buffer.concat(chunks).toString('utf-8')));
+	});
+}
+
+describe('LibraryStorage', () => {
+	let module: TestingModule;
+	let storage: LibraryStorage;
+	let s3ClientAdapter: DeepMocked<S3ClientAdapter>;
+	let repo: DeepMocked<LibraryRepo>;
+
+	beforeAll(async () => {
+		module = await Test.createTestingModule({
+			providers: [
+				LibraryStorage,
+				{
+					provide: LibraryRepo,
+					useValue: createMock<LibraryRepo>(),
+				},
+				{
+					provide: 'S3ClientAdapter_Libraries',
+					useValue: createMock<S3ClientAdapter>(),
+				},
+			],
+		}).compile();
+
+		storage = module.get(LibraryStorage);
+		s3ClientAdapter = module.get('S3ClientAdapter_Libraries');
+		repo = module.get(LibraryRepo);
+	});
+
+	afterAll(async () => {
+		await module.close();
+	});
+
+	beforeEach(() => {
+		jest.resetAllMocks();
+
+		const installedLibs: InstalledLibrary[] = [];
+
+		repo.getAll.mockImplementation(() => {
```

Comment:
Test Struktur anpassen und dabei auch die mock returns die jeweils benötigt werden in das setup, anstatt einmalig alles zu mocken.

Reply:
Test structure out of scope